File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
def solution(word):
    char_list = ['A', 'E', 'I', 'O', 'U']
    count = 0
    
    def dfs(current_word):
        nonlocal count
        
        count += 1
        if current_word == word:
            return True 
        if len(current_word) == 5: 
            return False  
        for char in char_list:
            if dfs(current_word + char):
                return True  
        return False
    
    dfs("")
    return count - 1  

File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
import java.util.*;

class Solution {
    int answer = 0;
    public int solution(String word) {
        StringBuilder currentWord = new StringBuilder();
        char[] alphabet = new char[]{'A', 'E', 'I', 'O', 'U'};
        dfs(word, currentWord, alphabet);
        return answer;
    }
    
    private boolean dfs(String word, StringBuilder currentWord, char[] alphabet) {
        System.out.println(currentWord);
        if(word.equals(currentWord.toString())) {
            return true;
        }
        if(currentWord.length() == 5) {
            return false;
        }
        
        
        for(char a : alphabet) {
            currentWord.append(a);
            answer++;
            if(dfs(word, currentWord, alphabet)){
                return true;
            };
            currentWord.deleteCharAt(currentWord.length() - 1);
        }
        
        return false;
    }
}